### DISCUSSION TITLE
I cant send email with bref

### DISCUSSION THREAD

**Discussion by oluwapaso** [brefphp/bref#1234] [General]
Hello, i tried using phpmailer on bref and keep getting connection error, then I tied using  native mail() function and I got this "sh: /usr/sbin/sendmail: No such file or directory". And to my knowledge phpmailer depends on mail() function. Please how can I solve this? i scan the /usr/sbin/ directory, sendmail don't come with amazon lambda container

**Comment by t-richard** [comment]
The native `mail` php function depends on the sendmail binary that is not bundled with Bref.

If you can, I would advise against using that and use a mail provider like AWS SES, Mailgun, Sendgrid, Postmark or similar services with `symfony/mailer` for example.

**Reply by oluwapaso** [reply]
Thanks, we have Sendgrid already we allow our users to integrate their own SMTP without paying extra. The aim is to get PHPMailer working though. I am currently routing the request to another Cpanel hosting to send the email until we find a solution.

**Reply by t-richard** [reply]
Are you using some abstraction library for Sendgrid ? I know libraries like symfony/mailer have SMTP transports that don't require sendmail (which is a different transport in symfony/mailer).

If PHPmailer is tighly coupled to sendmail, it seems like a bad fit for running in lambda.

You could always package sendmail in a lambda layer but it's probably a complex solution for something as simple as sending emails through SMTP.

**Reply by oluwapaso** [reply]
> Are you using some abstraction library for Sendgrid ? I know libraries like symfony/mailer have SMTP transports that don't require sendmail (which is a different transport in symfony/mailer).
> 
> If PHPmailer is tighly coupled to sendmail, it seems like a bad fit for running in lambda.
> 
> You could always package sendmail in a lambda layer but it's probably a complex solution for something as simple as sending emails through SMTP.

Thank you!. I have figured it out... PHPmailer is not tied to sendmail. It's SSL handshake issue. Normal PHPmailer $mail->SMTPDebug = SMTP::DEBUG_SERVER; is not reporting it. Had to use $mail->SMTPDebug = SMTP::DEBUG_LOWLEVEL; to see it.

I just had to turn off SSL check using

$mail->SMTPOptions = [
'ssl' => [
'verify_peer' => false,
'verify_peer_name' => false,
'allow_self_signed' => true,
]
];
